src/main/java/com/google/api/codegen/transformer/ApiMethodTransformer.java

```diff
@@ -44,6 +54,237 @@ public ApiMethodTransformer() {
+            .build();
+    methodViewBuilder.unpagedListCallableMethod(unpagedListCallableDetails);
+
+    methodViewBuilder.responseTypeName(
+        context.getTypeTable().getAndSaveNicknameFor(context.getMethod().getOutputType()));
+
+    return methodViewBuilder.type(ApiMethodType.UnpagedListCallableMethod).build();
+  }
+
+  public StaticApiMethodView generateFlattenedMethod(
+      MethodTransformerContext context, ImmutableList<Field> fields) {
+    StaticApiMethodView.Builder methodViewBuilder = StaticApiMethodView.newBuilder();
+
+    setCommonFields(context, methodViewBuilder);
+    methodViewBuilder.name(context.getNamer().getApiMethodName(context.getMethod()));
+    setFlattenedMethodFields(context, fields, methodViewBuilder);
+    setSyncStaticReturnFields(context, methodViewBuilder);
+
+    return methodViewBuilder.type(ApiMethodType.FlattenedMethod).build();
+  }
+
+  public StaticApiMethodView generateRequestObjectMethod(MethodTransformerContext context) {
+    SurfaceNamer namer = context.getNamer();
+    StaticApiMethodView.Builder methodViewBuilder = StaticApiMethodView.newBuilder();
+
+    setCommonFields(context, methodViewBuilder);
+    methodViewBuilder.name(namer.getApiMethodName(context.getMethod()));
+    setRequestObjectMethodFields(
+        context, namer.getCallableMethodName(context.getMethod()), methodViewBuilder);
+    setSyncStaticReturnFields(context, methodViewBuilder);
+
+    return methodViewBuilder.type(ApiMethodType.RequestObjectMethod).build();
+  }
+
+  public StaticApiMethodView generateCallableMethod(MethodTransformerContext context) {
+    SurfaceNamer namer = context.getNamer();
+    StaticApiMethodView.Builder methodViewBuilder = StaticApiMethodView.newBuilder();
+
+    setCommonFields(context, methodViewBuilder);
+    methodViewBuilder.name(namer.getCallableMethodName(context.getMethod()));
+    setCallableMethodFields(context, namer.getCallableName(context.getMethod()), methodViewBuilder);
+    methodViewBuilder.responseTypeName(
+        context.getTypeTable().getAndSaveNicknameFor(context.getMethod().getOutputType()));
+    methodViewBuilder.hasReturnValue(
+        !ServiceMessages.s_isEmptyType(context.getMethod().getOutputType()));
+
+    return methodViewBuilder.type(ApiMethodType.CallableMethod).build();
+  }
+
+  public void setCommonFields(
+      MethodTransformerContext context, StaticApiMethodView.Builder methodViewBuilder) {
+    SurfaceNamer namer = context.getNamer();
+    methodViewBuilder.apiRequestTypeName(
+        context.getTypeTable().getAndSaveNicknameFor(context.getMethod().getInputType()));
+    methodViewBuilder.apiClassName(namer.getApiWrapperClassName(context.getInterface()));
+    methodViewBuilder.apiVariableName(namer.getApiWrapperVariableName(context.getInterface()));
+    methodViewBuilder.settingsGetterName(namer.getSettingsFunctionName(context.getMethod()));
+  }
+
+  private void setListMethodFields(
```

Comment:
Hmm that wasn't really intentional. I think it is an artifact of how I generated the method. I will make more methods in this class private.